packages/angular/projects/ui-angular/src/lib/primitives-poc/fields/form-fields-component/form-fields.component.ts

```diff
@@ -0,0 +1,26 @@
+import { Component, ElementRef, Renderer2 } from '@angular/core';
+import { AmplifyBasePrimitiveComponent } from '../../base-primitive/base-primitive.component';
+
+const $COMPONENT_SELECTOR: string = 'amplify-form-fields';
+
+@Component({
+  selector: $COMPONENT_SELECTOR,
+  template: '<ng-content></ng-content>',
+})
+export class AmplifyFormFieldsComponent extends AmplifyBasePrimitiveComponent {
+  constructor(private renderer: Renderer2, private element: ElementRef) {
+    super();
+  }
+
+  ngOnInit() {
+    const randomId = this.getRandomId();
+    let val = 'amplify-' + randomId;
+    for (let ele in this.element.nativeElement.children) {
+      this.renderer.setProperty(ele, 'id', val);
+    }
+  }
+
+  getRandomId() {
+    return Math.floor(Math.random() * 6 + 1);
```

Comment:
Prefer using `nanoid` which we already have installed. Here's how you can use it for example:

https://github.com/aws-amplify/amplify-ui/blob/c6f5a36bc61a62c04a90e681d309a822150c1b48/packages/angular/projects/ui-angular/src/lib/components/authenticator/components/form-field/form-field.component.ts#L21

Reply:
Done.